app/controllers/finance/provider/invoices_controller.rb

```diff
@@ -12,6 +12,11 @@ class Finance::Provider::InvoicesController < Finance::Provider::BaseController
   def index
     @search = ThreeScale::Search.new(params[:search] || params)
     @invoices = collection.scope_search(@search).order_by(params[:sort], params[:direction]).paginate(paginate_params).decorate
+    @years = Invoice.where(provider_account_id: current_account.id)
```

Comment:
Some notes:

1. I think `YEAR(period)` only works for MySQL, PostgreSQL and Oracle have other functions to extract the year. I suggest using BabySqueel, that same way that `Invoice.years` is currently using - in fact, I think the easiest would be to modify that method to incorporate `provider_account_id` in there, as this method would only be used by this controller (if I am not mistaken)

2. Well, that's also included in 1 - I suggest to move this logic to the model, and not the controller.

3. The query should return the years in order, because the view queries for `first` and `last`, and if the years are not ordered, the result will not be correct.